src/zip.ts

```diff
@@ -176,7 +176,7 @@ export function centralHeader(file: ZipEntryDescription & Metadata, offset: bigi
   header.setUint16(30, zip64HeaderLength, true)
   // useless disk fields = zero (4 bytes)
   // useless attributes = zero (4 bytes)
-  header.setUint16(40, file.isFile ? 0o100664 : 0o040775, true) // UNIX regular file with permissions 664, or folder with permission 775.
+  header.setUint16(40, file.mode != null ? (file.mode | (file.isFile ? 0o100000 : 0o040000)) : (file.isFile ? 0o100664 : 0o040775), true) // default permissions: 664 for regular file, 775 for folder.
```

Comment:
I think it would be nicer and more consistent to set the default mode in `normalizeInput` at the same time that we determine if the file is a folder, so when you get to *this* point, `file.mode` is always set and you can OR it with `0o100000` or `0o040000` with a single ternary instead of three.

Reply:
PR updated as requested.